src/overlays/actors/ovl_En_Ice_Hono/z_en_ice_hono.c

```diff
@@ -21,37 +32,345 @@ const ActorInit En_Ice_Hono_InitVars = {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ICE_HONO, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0,
+                        ((s32)(Math_Rand_ZeroOne() * 1000.0f) + i) - 0x1F4, 0, 2);
+        }
+    }
+
+    if (this->timer <= 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnIceHono_SetupActionSmallFlame(EnIceHono* this) {
+    f32 temp_f0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/func_80A730BC.s")
+    this->actionFunc = EnIceHono_SmallFlameMove;
+    this->timer = 0x2C;
+    this->alpha = 0xFF;
+    if (this->actor.params == 1) {
+        this->smallFlameTargetYScale = (Math_Rand_ZeroOne() * 0.005f) + 0.004f;
+        this->actor.speedXZ = (Math_Rand_ZeroOne() * 1.6f) + 0.5f;
+    }
+    else
+    {
+        this->smallFlameTargetYScale = (Math_Rand_ZeroOne() * 0.005f) + 0.003f;
+        temp_f0 = Math_Rand_ZeroOne();
+        this->actor.speedXZ = (temp_f0 + temp_f0) + 0.5f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/func_80A731C0.s")
+void EnIceHono_SmallFlameMove(EnIceHono* this, GlobalContext* globalCtx) {
+    f32 unk = 10.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/EnIceHono_Init.s")
+    if (this->timer >= 0x15) {
+        Math_ApproxF(&this->actor.scale.x, 0.006f, 0.00016f);
+        Math_ApproxF(&this->actor.scale.y, this->smallFlameTargetYScale * 0.667f, 0.00014f);
+    } else {
+        Math_ApproxF(&this->actor.scale.x, 0.0001f, 0.00015f);
+        Math_ApproxF(&this->actor.scale.y, 0.0001f, 0.00015f);
+    }
+    this->actor.scale.z = this->actor.scale.x;
+    Math_ApproxF(&this->actor.speedXZ, 0, 0.06f);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, unk, unk, 0.0f, 5);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/EnIceHono_Destroy.s")
+    if (this->timer < 0x19) {
+        this->alpha -= 0xA;
+        this->alpha = CLAMP(this->alpha, 0, 0xFF);
+    }
+    if (this->timer <= 0) {
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/func_80A733E8.s")
+void EnIceHono_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+    f32 temp_f16;
+    f32 temp_f2;
+    f32 pad;
+    f32 sin154;
+    f32 sin156;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/func_80A734D4.s")
+    if (this->timer > 0) {
+        this->timer -= 1;
+    }
+    if (this->actor.params == 0) {
+        func_8002F8F0(&this->actor, 0x1048);
+    }
+    if ((this->actor.params == -1) || (this->actor.params == 0)) {
+        this->unk154 += 0x1111;
+        this->unk156 += 0x4000;
+        sin156 = Math_Sins(this->unk156);
+        sin154 = Math_Sins(this->unk154);
+        temp_f16 = (Math_Rand_ZeroOne() * 0.05f) + ((sin154 * 0.125f) + (sin156 * 0.1f));
```

Comment:
unk154 and 156 are part of making the light color change over time, by computing its intensity
may want to call them "colorTimer1" and "colorTimer2" or something like that?
if you keep unk the naming is `unk_XXX`

Reply:
They are not really timers though, right? I kept them `unk_XXX` for now.

Reply:
they're angles being added the same amount every frame and used for animating so I would consider them timers, but maybe it's a bit of a stretch indeed 🤷